### ISSUE TITLE
Check all required types are exported

### ISSUE THREAD

**Issue by garyb** [purescript/purescript#959]
It seems there is still a problem with this, discovered when I was hacking down QC:

``` purescript
module Test.QuickCheck.Gen
  ( Gen()
  , uniform
  , perturbGen
  , repeatable

  , GenState()
  , GenOut()
  , runGen
  , evalGen

  -- it should be an error not to export these, as they're used in `GenState`
  -- but it's not an error until trying to recompile as they're missing from the externs
  -- , Size()
  -- , LCG()

  , stateful
  ) where
```

**Comment by paf31** [comment]
Possibly the same as #1179, #1318?

**Comment by garyb** [comment]
Not quite - this didn't generate invalid externs, it just worked even though in theory there's a check to disallow it.

I'm thinking maybe we should remove the check now though, or relax it, as there are actually times where not exporting a type that is used in a type synonym is actually perfectly okay.

**Comment by garyb** [comment]
Irrelevant as of #1525

**Comment by garyb** [comment]
This was resolved in #1529. Still want to revisit the idea of being able to use "abstract type synonyms", but at least this particular check works as it should now.
